### PR TITLE
Custom basler images

### PR DESCRIPTION
**Pull request by thomaswmorris**
and also a few bug fixes

### CODE REVIEW

**Review by mrakitin — COMMENTED** [review]
Thanks, @thomaswmorris!
I think it needs more work. Please see my comments below.

Also, could you please upload the generated images as artifacts via GitHub Actions? See an example [here](https://github.com/NSLS-II/sirepo-bluesky/blob/60c532a6bfaf7558a2ec7443a63fcec0b3229ad0/.github/workflows/publish-docs.yml#L92-L95).

File: ophyd_basler/custom_images.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+def gaussian(x, y, a, cx, cy, sx, sy):
+    return a * np.exp(-0.5 * (np.square((x - cx) / sx) + np.square((y - cy) / sy)))
+
+
+def get_wandering_gaussian_beam(nf, nx, ny):
+
+    # hard-coded for now
+
+    length_scale = 64
+    bounds = np.array([[0, 256], [0, nx], [0, ny], [nx / 64, nx / 8], [ny / 64, ny / 8]])
```
[the remark this answers is not in the document]
Comment (thomaswmorris) [reply ophyd_basler/custom_images.py:13]: This is a different parameter (it's the time length scale). I renamed it "time_scale" to avoid confusion.

**Comment by mrakitin** [comment]
You could install the pre-commit hook into git, and it will run it automatically before each commit. Check https://pre-commit.com/#3-install-the-git-hook-scripts for details.

**Comment by thomaswmorris** [comment]
There is some weird overhead going on with the first acquisition; I changed the test so that it still tests the duration (but is more robust).

**Comment by mrakitin** [comment]
I think we can exclude the first duration, and increase the overhead time from 0.1 to a larger value, but your solution looks good too.

Another observation: I tried to run the code, and on every count, it returns the same frame. Is it possible to continue with the next random image rather than starting from the very beginning?

**Comment by thomaswmorris** [comment]
> Another observation: I tried to run the code, and on every count, it returns the same frame. Is it possible to continue with the next random image rather than starting from the very beginning?

Is bluesky staging and unstaging on every count? Probably every time we do camera.Open() it restarts the same generated images. Does this happen for the default emulated images (the stripes)?

**Comment by mrakitin** [comment]
> Is bluesky staging and unstaging on every count?

No, it does it only once per scan.

> Probably every time we do camera.Open() it restarts the same generated images. Does this happen for the default emulated images (the stripes)?

Yes, it did happen for the stripes too. I have found a solution for that, will post as a separate PR to this PR branch.

**Review by mrakitin — APPROVED** [review]

**Pull request merged**